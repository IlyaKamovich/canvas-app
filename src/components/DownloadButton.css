.download-button {
  margin-bottom: 1rem;
}
