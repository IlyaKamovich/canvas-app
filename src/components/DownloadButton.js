import React from 'react';

import styles from './DownloadButton.module.scss';

function DownloadButton(props) {
  return (
    <div className={styles.download__container}>
      <a href="/" className={styles.download__button}>
        Download this canvas
      </a>
    </div>
  );
}

export default DownloadButton;
