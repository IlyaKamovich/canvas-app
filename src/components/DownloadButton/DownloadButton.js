import React from 'react';

import styles from './DownloadButton.module.scss';
import { useAppState } from '../../AppProvider';

function DownloadButton() {
  const { url: href } = useAppState();
  const Component = href ? 'a' : 'button';

  return (
    <div className={styles.download__container}>
      <Component
        href={href}
        type={!href ? 'button' : null}
        className={styles.download__button}
        download={href ? 'output' : null}
        disabled={!href}
      >
        Download this canvas
      </Component>
    </div>
  );
}

export default DownloadButton;
