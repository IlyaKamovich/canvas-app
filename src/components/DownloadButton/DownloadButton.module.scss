@value inputDataFieldWidth from "../DateFileInput/DataFileInput.module.scss";

.download__container {
  width: 100%;
  display: inline-block;
  text-align: center;
}

@value downloadButtonHeight: 4.5rem;

.download__button {
  margin-top: 2rem;
  display: inline-block;
  width: calc(inputDataFieldWidth - 10rem);
  height: downloadButtonHeight;
  line-height: downloadButtonHeight;
  border: none;
  color: #fff;
  background: #16a085;
  border-radius: 4px;
  border-bottom: 4px solid #117a60;
  transition: all 0.5s ease;
  outline: none;
  user-select: none;

  &,
  &:visited {
    text-transform: uppercase;
    text-decoration: none;
  }

  &:hover {
    background: #149174;
    color: #0c5645;
    cursor: pointer;

    transform: scale(1.05);
  }

  &:active {
    border: 0;

    transform: translateY(10%) scale(1.05);
  }

  &[disabled] {
    opacity: 0.8;
    cursor: not-allowed;
  }

  &:focus {
    transform: translateY(10%) scale(1.05);
    border: none;
    outline: 2px dotted #117a60;
    outline: 2px solid -webkit-focus-ring-color;
  }
}
