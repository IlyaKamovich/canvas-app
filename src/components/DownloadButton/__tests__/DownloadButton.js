import React from 'react';
import { render } from '@testing-library/react';
import DownloadButton from '../DownloadButton';

describe('<DataFileInput /> spec', () => {
  it('renders the component', () => {
    const { container } = render(<DownloadButton />);
    expect(container).toMatchSnapshot();
  });
});
