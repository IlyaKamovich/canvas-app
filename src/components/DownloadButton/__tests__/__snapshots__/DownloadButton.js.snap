// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataFileInput /> spec renders the component 1`] = `
<div>
  <div
    class="download__container"
  >
    <a
      class="download__button"
      href="/"
    >
      Download this canvas
    </a>
  </div>
</div>
`;
